pyomo/contrib/alternative_solutions/tests/test_balas.py

```diff
@@ -0,0 +1,145 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2024
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from collections import Counter
+
+from pyomo.common.dependencies import numpy as numpy, numpy_available
+
+if numpy_available:
+    from numpy.testing import assert_array_almost_equal
+
+import pyomo.environ as pe
+from pyomo.common import unittest
+import pyomo.opt
+
+from pyomo.contrib.alternative_solutions import enumerate_binary_solutions
+import pyomo.contrib.alternative_solutions.tests.test_cases as tc
+
+
+solvers = list(pyomo.opt.check_available_solvers("glpk", "gurobi", "appsi_gurobi"))
+pytestmark = unittest.pytest.mark.parametrize("mip_solver", solvers)
+
+
+@unittest.pytest.mark.default
+class TestBalasUnit:
```

Comment:
Generally we do this like it is in `test_aos_utils.py` where we inherit from `unittest.TestCase` and then you can use the different types of assertion methods that come on that base class rather than just `assert` statements. Not sure if it needs to hold up this PR, but it would be preferable to use that style since failures will be much easier to interpret. (I think this comment holds for many of the test files below as well.)

Reply:
This won't be a simple change given the pytest logic I'm leveraging to parameterize tests.  Let's do this after the PR.